### DISCUSSION TITLE
[General] Withdraw contract balance

### DISCUSSION THREAD

**Discussion by nicob21** [zkSync-Community-Hub/zksync-developers#229] [General] [answered]
### Environment

Mainnet

### Provide a brief description of the functionality you're trying to implement.

We've deployed a [contract](https://explorer.zksync.io/address/0x6E84Bbd27946227f19A07Af113c2Ff84B288B152) so that our users can mint an NFT on zkSync Era but we can't withdraw the funds from the contract.

### What is the specific issue or error you're encountering?

The withdraw function is failing to execute:
```
function withdraw() external onlyOwner {
        payable(owner()).transfer(address(this).balance);
}
```

### Can you share the error messages or logs you're receiving, if any?

.

### Have you made any recent changes to the contract before encountering this issue?

No

### Are there any external libraries or contracts that your contract interacts with?

No

### Can you provide the relevant portions of your contract code where the issue is occurring?

```
function withdraw() external onlyOwner {
        payable(owner()).transfer(address(this).balance);
}
```

### Have you tried to isolate the problem, and if so, what were the results?

Using `call` instead of `transfer` fixed the issue but the funds are stuck in the deployed contract...

### What steps have you already taken to try to resolve the issue?

I've replaced the `withdraw` function with to:
```
function withdraw() external onlyOwner {
        (bool success,) = payable(owner()).call{value: address(this).balance}("");
        require(success, "FAILED_TO_WITHDRAW");
}
```
and deployed a new contract.

It's similar to this issue where you found a way to help rescue the funds... 
https://medium.com/coinmonks/gemstoneido-contract-stuck-with-921-eth-an-analysis-of-why-transfer-does-not-work-on-zksync-era-d5a01807227d

Sorry, our mistake, but wondering if it could be possible in our case too!

**Comment by vasyl-ivanchuk** [comment, marked as answer]
Hi @nicob21 👋 ,

It is a known limitation, please check our [official docs](https://era.zksync.io/docs/dev/building-on-zksync/best-practices.html#use-call-over-send-or-transfer) for more info. We are actively working to address this issue, but until then there is nothing we can really do.

**Reply by nicob21** [reply]
Thanks for your reply, so there is a chance to recover the funds when you address the issue?

**Reply by vasyl-ivanchuk** [reply]
Yes, correct!

**Comment by inpaslimabj** [comment]
It seems like your contract might be encountering issues with the `transfer` function on zkSync due to limitations in its handling of external calls. The approach you've taken by using `call` instead is a common workaround to ensure successful fund withdrawal.

To rescue the funds stuck in the contract, you can follow a similar approach to the one described in the linked article. This involves deploying a new contract with the corrected withdrawal function and transferring the ownership and funds from the old contract to the new one using a manual recovery process.

Be sure to thoroughly test the new contract and recovery process before proceeding to ensure that funds are safely transferred without any unexpected issues.

**Reply by nicob21** [reply]
Thanks for your reply, could you specify which linked article your are referring to? Thanks

**Comment by batesb188** [comment]
Encountered an issue with the withdraw function failing to execute on zkSync Era. Replaced transfer with call, but funds are stuck in the contract. Considering solutions similar to the Coinmonks article to rescue the funds.

**Comment by lynnrosado** [comment]
It seems like the issue might be related to using `transfer` instead of `call` for withdrawing funds from the contract. You can try using `call` with a require statement to handle potential failures, similar to the approach mentioned in the linked article.

**Comment by tcaniao** [comment]
It seems that using `.call` instead of `.transfer` resolved the issue, but the funds are still stuck in the contract. 
To further investigate and potentially rescue the funds, consider reaching out to zkSync support or consulting with a blockchain developer familiar with zkSync contracts.

**Comment by crissshawn** [comment]
The user has deployed a contract for minting NFTs on zkSync Era, but is unable to withdraw the funds from the contract using the transfer function. They have replaced the withdraw function with a new one that uses call instead of transfer, and have deployed a new contract. They are wondering if it is possible to recover the funds from the original contract, and have referenced a similar issue where a solution was found. No repository link was provided.
